File: articles/ai-agents-tool-use-and-procedural-autonomy/cpp/agent_permission.cpp
#include <iostream>
#include <string>

int main() {
    double risk = 0.85;
    bool approval_required = true;
    bool approved = false;
    std::string status = "pass";
    if (approval_required && !approved) status = "blocked";
    else if (risk >= 0.65) status = "escalate";
    std::cout << "agent_action_status=" << status << "\n";
    return 0;
}

File: articles/ai-agents-tool-use-and-procedural-autonomy/rust/agent_permission.rs
fn main() {
    let risk = 0.85;
    let approval_required = true;
    let approved = false;
    let status = if approval_required && !approved {
        "blocked"
    } else if risk >= 0.65 {
        "escalate"
    } else {
        "pass"
    };
    println!("agent_action_status={}", status);
}
